trunk/SUAVE/Methods/Noise/Fidelity_One/Noise_Tools/SPL_harmonic_to_third_octave.py

```diff
@@ -0,0 +1,67 @@
+## @ingroupMethods-Noise-Fidelity_One-Noise_Tools
+# SPL_harmonic_to_third_octave.py
+# 
+# Created:  Jul 2015, C. Ilario
+# Modified: Jan 2016, E. Botero
+# Modified: Apr 2021, M. Clarke
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+
+import numpy as np
+from SUAVE.Methods.Noise.Fidelity_One.Noise_Tools import SPL_arithmetic
+# ----------------------------------------------------------------------
+#  dbA Noise
+# ----------------------------------------------------------------------
+
+## @ingroupMethods-Noise-Fidelity_One-Noise_Tools 
+def SPL_harmonic_to_third_octave(SPL,f,settings): 
+    """This method converts the SPL spectrum from blade harmonic passing frequency
+    to thrid octave spectrum
+    
+    Assumptions:
+        N/A
+
+    Source: 
+
+    Inputs:
+        SPL                    - sound pressure level                          [dB] 
+        f                      - blade passing spectrum frequencies            [Hz]
+        settings.    
+            center_frequencies - center frequencies of the 1/3 octave spectrum [dB]  
+            lower_frequencies  - lower frequencies of the 1/3 octave spectrum  [dB]
+            upper_frequencies  - upper frequencies of the 1/3 octave spectrum  [dB]
+        
+
+    Outputs:
+        SPL_third_octave       - SPL in the 1/3 octave spectrum                [dB] 
+
+    Properties Used:
+        N/A 
+        
+    """  
+    # unpack 
+    cf               = settings.center_frequencies
+    lf               = settings.lower_frequencies
+    uf               = settings.upper_frequencies
+    
+    dim_cpt          = len(SPL[:,0,0,0])
+    dim_mic          = len(SPL[0,:,0,0])
+    dim_prop         = len(SPL[0,0,:,0])
+    num_cf           = len(cf)
+    num_f            = len(f)
+    SPL_third_octave = np.zeros((dim_cpt,dim_mic,dim_prop,num_cf)) 
+    
+    # loop through 1/3 octave spectra and sum up components 
```

Comment:
I actually vectorized this as much as it can be vectorized - Spoke to Emilio about this.. This is. Converting to/from different sound spectra required you to out each band in buckets...which change depending on the time, source.